src/world/Units/Players/Player.Legacy.cpp

```diff
@@ -860,26 +860,39 @@ bool Player::Create(CharCreate& charCreateContent)
     //setScale( ((race==RACE_TAUREN)?1.3f:1.0f));
     setScale(1.0f);
     setHealth(info->health);
-    setPower(POWER_TYPE_MANA, info->mana);
```

Comment:
I don't think removing these lines is correct.

Every character still has those power types, they should just be 0 instead (e.g. for a warrior's mana). It might be initialised to 0 elsewhere, but I think it makes sense to still set it to whatever is in the info struct and just make sure the info struct has the right data.

I don't have a working AscEmu build at the moment so I can't check, but it should be remarked on either way.

Reply:
They're getting overriden by next SetPower call for sure if that next power isn't equal to zero, warriors or thers i'm sure they're not supposed to have other types. My intended change is - to reduce setPowerType calls, instead of recalling and overwritting existing powers, it can be called only once.

Reply:
I agree with sanctum in the current situation.